File: Arquivos JavaScript/Ad-hoc/1793_Escada Rolante.js
let input = require("fs").readFileSync("stdin", "utf8");
let lines = input.split("\n");

for (let i = 1; i < lines.length; i+=2){
    let lista = (lines[i].split(" ")).map((e) => parseInt(e));
    let soma = 10;
    let tam = lista.length-1;
    let ant = lista[tam];
   
    if (lines[i-1] == 0){
        break;
    }
    
    for (let i = tam-1; i >= 0 ; i--){
        let calc = ant-lista[i];

        if (calc >= 10){
            soma+=10;
        }
        else{
            soma+=calc;
        }
        ant = lista[i];
    }
    console.log(soma);       
}

File: Arquivos Python/Ad-Hoc/1793_Escada Rolante.py
while(True):
    n = int(input())
    if n == 0:
        break
    t = input().split(" ")
    soma = 10
    ant = int(t[-1])

    for j in range(n, 0, -1):
        calc = ant-int(t[j-1])

        if calc >= 10:
            soma+=10
        else:
            soma+=calc
        ant = int(t[j-1])
    print(soma)
    
        